File: bindings/java-jni/src/main/java/org/maplibre/nativejni/error/AbiVersionMismatchException.java
package org.maplibre.nativejni.error;

/** Thrown when the loaded native library exposes an unsupported C ABI version. */
public final class AbiVersionMismatchException extends MaplibreException {
  private final int actualVersion;
  private final int expectedVersion;

  public AbiVersionMismatchException(int actualVersion, int expectedVersion) {
    super(
        MaplibreStatus.NATIVE_ERROR,
        0,
        "Unsupported Maplibre C ABI version %d; expected %d"
            .formatted(actualVersion, expectedVersion));
    this.actualVersion = actualVersion;
    this.expectedVersion = expectedVersion;
  }

  public int actualVersion() {
    return actualVersion;
  }

  public int expectedVersion() {
    return expectedVersion;
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/error/AbiVersionMismatchException.kt
package org.maplibre.nativeffi.error

/** Error for a loaded native library with an unsupported C ABI version. */
public class AbiVersionMismatchException(
  public val actualVersion: Long,
  public val expectedVersion: Long,
) :
  NativeErrorException(
    MaplibreStatus.NATIVE_ERROR.nativeCode,
    "Unsupported Maplibre C ABI version $actualVersion; expected $expectedVersion",
  )
